components/director/internal/domain/application/resolver.go

```diff
@@ -238,6 +254,62 @@ func (r *Resolver) Application(ctx context.Context, id string) (*graphql.Applica
 	return r.appConverter.ToGraphQL(app), nil
 }
 
+// ApplicationForTenant could be executed only in a certificate flow. It fetches model.ApplicationTemplate by label scenarioassignment.SubaccountIDKey
+// with value equal to the consumer ID. Lists all tenant header scoped []model.Application and finds the one created by the previously fetched App Template
+func (r *Resolver) ApplicationForTenant(ctx context.Context) (*graphql.Application, error) {
+	consumerInfo, err := consumer.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if !consumerInfo.Flow.IsCertFlow() {
+		return nil, apperrors.NewInvalidOperationError("query is accessible only with а certificate authentication")
+	}
+
+	tx, err := r.transact.Begin()
+	if err != nil {
+		return nil, err
+	}
+	defer r.transact.RollbackUnlessCommitted(ctx, tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	filters := []*labelfilter.LabelFilter{
+		labelfilter.NewForKeyWithQuery(scenarioassignment.SubaccountIDKey, fmt.Sprintf("\"%s\"", consumerInfo.ConsumerID)),
```

Comment:
Also add `region` and `xsappname` as label filters here. To do that:
1. Add two more fields - Region and TokenClientID (it contains the xsappname) to the [Consumer type](https://github.com/kyma-incubator/compass/blob/896cfbcca39fdf402cb0d82118be2dbc841379f4/components/director/pkg/consumer/consumer.go#L28):
```
	Region        string                `json:"region"`
	TokenClientID string                `json:"tokenClientID"`
```
2. Make sure to assign it to the consumerInfo context before saving it to the context [HERE](https://github.com/kyma-incubator/compass/blob/8a386b80908a4b0d1c1c2d4384a837a4e455d9b2/components/director/internal/authenticator/claims/claims.go#L78)
3. Make use of it here (you have access to the consumerInfo object)

> Note: The `consumerInfo.TokenClientID` field is not exactly equal to the xsappname. The xsappname is part of the TokenClientID string - more specifically it's "<some-prefix>-<xsappname>". There is a configuration property for what this prefix is (it is known to us), so check [HERE](https://github.com/kyma-incubator/compass/blob/896cfbcca39fdf402cb0d82118be2dbc841379f4/components/director/internal/authenticator/claims/validator.go#L106) for reference. Only after you trim the prefix can you use the resulting string as the xsappname label filter value.